koap/src/jsMain/kotlin/adapter/Message.kt

```diff
@@ -0,0 +1,181 @@
+package com.juul.koap.adapter
+
+import com.juul.koap.Message
+import com.juul.koap.adapter.Message as AdaptedMessage
+import com.juul.koap.adapter.Code as AdaptedCode
+import com.juul.koap.adapter.Option as AdaptedOption
+import com.juul.koap.adapter.Option.OptionType as AdaptedOptionType
+
+
+internal fun Message.toAdapted(): AdaptedMessage = when (this) {
+    is Message.Tcp -> {
+        AdaptedMessage(
+            messageType = MessageType.TCP,
+            code = code.toAdapted(),
+            token = token,
+            options = options.map { it.toAdapted() },
+            payload = payload.decodeToString()
+        )
+    }
+    is Message.Udp -> {
+        AdaptedMessage(
+            messageType = MessageType.UDP,
+            code = code.toAdapted(),
+            type = type.name,
+            id = id,
+            token = token,
+            options = options.map { it.toAdapted() },
+            payload = payload.decodeToString()
+        )
+    }
```

Comment:
nitpick: Unnecessary curly braces are just noise.

```suggestion
    is Message.Tcp -> AdaptedMessage(
        messageType = MessageType.TCP,
        code = code.toAdapted(),
        token = token,
        options = options.map { it.toAdapted() },
        payload = payload.decodeToString()
    )
    is Message.Udp -> AdaptedMessage(
        messageType = MessageType.UDP,
        code = code.toAdapted(),
        type = type.name,
        id = id,
        token = token,
        options = options.map { it.toAdapted() },
        payload = payload.decodeToString()
    )
```

Reply:
Fixed